core/math/a_star.cpp

```diff
@@ -280,10 +280,13 @@ int AStar::get_closest_point(const Vector3 &p_point, bool p_include_disabled) co
 			continue; // Disabled points should not be considered.
 		}
 
+		// Keep the closest point's ID, and in case of multiple closest IDs,
+		// the smallest one (makes it deterministic).
 		real_t d = p_point.distance_squared_to((*it.value)->pos);
-		if (closest_id < 0 || d < closest_dist) {
+		int id = *(it.key);
+		if (closest_id < 0 || d < closest_dist || (d == closest_dist && id < closest_id)) {
 			closest_dist = d;
-			closest_id = *(it.key);
+			closest_id = id;
 		}
 	}
```

Comment:
Usually for this kind of thing, there's no need to check the closest_id < 0, because any distance will always be smaller than FLT_MAX.

The d < closest_dist and (d == closest_dist) are usually better combined then separated with a later check, i.e.
```
if (d <= closest_dist)
{
    if (d == closest_dist) ....
}
```

Reply:
> Usually for this kind of thing, there's no need to check the closest_id < 0, because any distance will always be smaller than FLT_MAX.

That's used for the first run as `closest_id` is initialized to `-1`. I thought about moving the first `it = points.iter()` out of the `for` to set the initially value to something valid, but then it would require also check that the point is enabled and that there's any point at all, etc., so in the end I guess there's no clean way around it.

Reply:
Or nevermind, I understand what you meant now. Indeed if we enter the loop, we'll override the value on the first run, so the check is redundant.

Reply:
> The d < closest_dist and (d == closest_dist) are usually better combined then separated with a later check, i.e.

That wouldn't really work here, I'd have to do an early continue instead:
```
		if (d <= closest_dist) {
			if (d == closest_dist && id > closest_id)) { // Keep lowest ID.
				continue;
			}
			closest_dist = d;
			closest_id = id;
		}
```

Reply:
Pushed an amended commit.